File: client/src/app/models/product.ts
export interface Product {
  id: number;
  nombre: string;
  precio: number;
  categoria?: string;
  descripcion?: string;
  imagen?: string;    // base64 o URL
  estado: boolean;
}

export interface ProductCreateDto {
  nombre: string;
  precio: number;
  categoria?: string;
  descripcion?: string;
  estado: boolean;
  imagenFile?: File;
}

export interface ProductUpdateDto extends ProductCreateDto {}

File: PruebaTecnicaCycle/PruebaTecnicaCycle.Domain/Entities/Product.cs
namespace PruebaTecnicaCycle.Domain.Entities
{
    public class Product
    {
        public int Id { get; set; }
        public string Nombre { get; set; } = null!;
        public decimal Precio { get; set; }
        public string? Categoria { get; set; }
        public string? Descripcion { get; set; }
        public string? Imagen { get; set; }  // Base64 o URL
        public bool Estado { get; set; }
    }
}
